src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorSubmitApplication.java

```diff
@@ -139,6 +130,24 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
                 .build();
         }
 
+        final var application = caseData.getApplication();
+        final var applicationFeeOrderSummary = application.getApplicationFeeOrderSummary();
+
+        if (caseData.getApplication().isSolicitorPaymentMethodPba()) {
+            PbaResponse response = paymentService.processPbaPayment(caseData, caseId);
+
+            if (response.getHttpStatus() == CREATED) {
+                updateCaseDataWithPaymentDetails(applicationFeeOrderSummary,caseData, response.getPaymentReference());
+            } else {
+                return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                    .data(details.getData())
+                    .errors(singletonList(response.getErrorMessage()))
+                    .build();
+            }
+        }
```

Comment:
Shouldn't there be an else statement here? I guess the only other option is HWF?

Reply:
yes we have only two options so it's either paid via PBA or HWF

Reply:
Ah, I guess that happens in the case submission tasks